src/run_encoded_array.cpp

```diff
@@ -0,0 +1,73 @@
+#include "sparrow/layout/run_end_encoded_layout/run_end_encoded_array.hpp"
+#include "sparrow/layout/primitive_array.hpp"
+#include "sparrow/layout/dispatch.hpp"
+#include "sparrow/layout/dispatch_lib.hpp"
+
+
+
+namespace sparrow   
+{
+    template<class T>
+    struct is_usable_array : std::false_type
+    {
+    };
+    template<>
+    struct is_usable_array<primitive_array<std::uint16_t>> : std::true_type
+    {
+    };
+    template<>
+    struct is_usable_array<primitive_array<std::uint32_t>> : std::true_type
+    {
+    };
+    template<>
+    struct is_usable_array<primitive_array<std::uint64_t>> : std::true_type
+    {
+    };
+
+
+    
+    
+
+    auto run_end_encoded_array::get_acc_lengths_ptr(const array_wrapper& ar) -> acc_length_ptr_variant_type
+    {
+
+        return visit(
+            [](const auto& actual_arr) -> acc_length_ptr_variant_type
+            {
+                using array_type = std::decay_t<decltype(actual_arr)>;
+                if constexpr( is_usable_array<array_type>::value)
+                {
+                    return actual_arr.data();
+                }
+                else
+                {
+                    throw std::invalid_argument("array type not supported");
```

Comment:
To replace with a static assert

Reply:
this is not working...even with a dependent false this will aways result in a compile-time-error

Reply:
It's really unclear to my why, sometimes, a simple "else" is enough, sometimes we have to use a "dependent false", and sometimes nothing works.
Have you tried to do an "else if (!is_usable_array ...)"

Reply:
so a non-templated (library) function, I believe it will never work because its "fully evaluated"

Reply:
Ho you right, I didn't noticed that